util/patch/patch.go

```diff
@@ -291,6 +296,15 @@ func (h *Helper) calculateChanges(after client.Object) (map[string]bool, error)
 		return nil, errors.Wrapf(err, "failed to unmarshal patch data into a map")
 	}
 
+	// log whether a patch is created. If enabled log the full patch.
+	if len(patchDiff) == 0 {
+		log.Info(fmt.Sprintf("No changes for %s", tlog.KObj{Obj: h.after}))
```

Comment:
Should we add the object as k/v pair to the logger too (as it's not necessarily already there) or will we address this when we address it holistically?

Reply:
Good idea.

Reply:
Problem is we have to already use klog.KObj because otherwise the object won't be logged correctly with JSON and text. So I'm absolutely fine to defer or choose some non-controversial option for this PR.

Reply:
Lets defer for now - assuming that we hit most of the cases by adding the k/v pair in the reconcilers / controller runtime.